lib/experimental/interactivity-api/directive-processing.php

```diff
@@ -28,37 +28,255 @@ function gutenberg_interactivity_mark_root_blocks( $parsed_block, $source_block,
+ * @param WP_Block $block_instance The block instance.
+ */
+function gutenberg_mark_block_interactivity( $block_content, $block, $block_instance ) {
+	if (
+		isset( $block_instance->block_type->supports['interactivity'] ) &&
+		$block_instance->block_type->supports['interactivity']
+	) {
+		// Wraps the interactive block with a comment delimiter to be able to
+		// process it later.
+		return get_comment_delimited_block_content(
+			'core/interactivity-wrapper',
+			array(),
+			$block_content
+		);
+	} elseif ( WP_Directive_Processor::is_marked_as_children_of_interactive_block( $block ) ) {
+		WP_Directive_Processor::unmark_children_of_interactive_block( $block );
 
+		// Wraps the non-interactive block with a comment delimiter to be able to
+		// skip it later.
+		return get_comment_delimited_block_content(
+			'core/non-interactivity-wrapper',
+			array(),
+			$block_content
+		);
+	}
+	return $block_content;
+}
+add_filter( 'render_block', 'gutenberg_mark_block_interactivity', 10, 3 );
+
+/**
+ * Traverses the HTML of an interactive block, searching for Interactivity API
+ * directives and processing them. For the inner blocks, it calls the
+ * corresponding function depending on the wrapper type.
+ *
+ * @param array                $interactive_block The interactive block to process.
+ * @param WP_Directive_Context $context The context to use when processing.
+ * @param array                $directives The directives to apply.
+ *
+ * @return string The processed HTML.
+ */
+function gutenberg_process_interactive_block( $interactive_block, $context, $directives ) {
+	$block_index              = 0;
+	$content                  = '';
+	$interactive_inner_blocks = array();
+
+	foreach ( $interactive_block['innerContent'] as $inner_content ) {
+		if ( is_string( $inner_content ) ) {
+			$content .= $inner_content;
+		} else {
+			// This is an inner block. It may be an interactive block or a
+			// non-interactive block.
+			$content                   .= '<wp-inner-blocks-' . $block_index . '></wp-inner-blocks-' . $block_index . '>';
+			$interactive_inner_blocks[] = $interactive_block['innerBlocks'][ $block_index++ ];
+		}
+	}
+
+	return gutenberg_process_interactive_html( $content, $context, $directives, $interactive_inner_blocks );
+}
 
 /**
- * Resolve the reference using the store and the context from the provided path.
+ * Returns the HTML of a non-interactive block without processing the
+ * directives. For the inner blocks, it calls the corresponding function
+ * depending on the wrapper type.
+ *
+ * @param array                $non_interactive_block The non-interactive block to process.
+ * @param WP_Directive_Context $context The context to use when processing.
+ * @param array                $directives The directives to apply.
+ *
+ * @return string The processed HTML.
+ */
+function gutenberg_process_non_interactive_block( $non_interactive_block, $context, $directives ) {
```

Comment:
This function is not being called in examples that should be 😩 
```html
<!-- wp:group {"layout":{"type":"default"}} -->
  <div class="wp-block-group">
     <!-- wp:create-block/context -->
         <div class="wp-block-create-block-context">
              <!-- wp:paragraph -->
                <p>paragraph</p>
              <!-- /wp:paragraph -->
          </div>
      <!-- /wp:create-block/context -->
   </div>
<!-- /wp:group -->```

In that case the paragraph should be wrapped by `core/non-interactivity`.